server/src/utils/socketIoUtils.ts

```diff
@@ -61,20 +64,23 @@ export const setSubscribedRoadmap = (socket: ExtendedSocket) => async (
     return;
   }
 
-  socket.rooms.forEach((room) => socket.leave(room));
+  socket.rooms.forEach((room) => {
+    if (room.startsWith('roadmap-')) {
+      socket.leave(room);
+    }
+  });
   socket.join(`roadmap-${roadmapId}`);
-  // Sockets are added to a room with their user id so we can emit events using user id
-  socket.join(`${socket.data.user!.id}`);
 
   sendResponse?.({ status: 200 });
 };
 
 export type EmitRoadmapEventParams<T extends ClientEvents> = {
   roadmapId: number;
-  dontEmitToUserId?: number;
-  requirePermission?: number;
   event: T;
   eventParams: Parameters<ClientEventsMap[T]>;
+  dontEmitToUserId?: number;
+  onlyEmitToUserId?: number;
```

Comment:
```suggestion
  onlyEmitToUserIds?: number[];
```

This could be an array